cavern/src/main/java/org/opencadc/cavern/files/PutAction.java

```diff
@@ -183,15 +185,53 @@ public void doAction() throws Exception {
 
             // doing this last because it requires chown which is most likely to fail during experimentation
             log.debug("restore owner & group");
-            UserPrincipal owner = NodeUtil.getOwner(getUpLookupSvc(), node);
-            GroupPrincipal group = NodeUtil.getDefaultGroup(getUpLookupSvc(), owner);
-            NodeUtil.setPosixOwnerGroup(rootPath, target, owner, group);
-            
+            restoreOwnNGroup(rootPath, node);
         } catch (AccessControlException | AccessDeniedException e) {
             log.debug(e);
             syncOutput.setCode(403);
+        } catch (IOException e) {
+            if (e.getMessage().contains("quota exceeded")) {
+                // TODO: Replace the quota limit below with the number of bytes 
+                //       remaining in the quota when we can determine it.
+                // get quota limit from a parent container
+                Path rootPath = Paths.get(getRoot());
+                restoreOwnNGroup(rootPath, node);
+                Node curNode = node.getParent();
+                String limitValue = curNode.getPropertyValue(VOS.PROPERTY_URI_QUOTA);
+                while (limitValue == null) {
+                    curNode = curNode.getParent();
+                    try {
+                        limitValue = curNode.getPropertyValue(VOS.PROPERTY_URI_QUOTA);
+                    } catch (NullPointerException ex) {
+                        // quota limit not defined in a parent container, exit while loop
```

Comment:
Hmm, NullPointerException means an error or bug and shouldn't be part of the program control.  What is null?  Could you check for null instead?

Reply:
Updated code to look for a null node and log a warning.